### ISSUE TITLE
Why are you comparing a number to an array?

### ISSUE THREAD

**Issue by alexandrucancescu** [victorqribeiro/MLP#1]
Making sure `x` is an array
https://github.com/victorqribeiro/MLP/blob/d6e50e2a3206e4d6233cc6c1a88d8203368ccb6f/js/MLP.js#L194-L195

Comparing `x` (array) to `i` (a number)
https://github.com/victorqribeiro/MLP/blob/d6e50e2a3206e4d6233cc6c1a88d8203368ccb6f/js/MLP.js#L198

Also, you are subtracting an array from a number

https://github.com/victorqribeiro/MLP/blob/d6e50e2a3206e4d6233cc6c1a88d8203368ccb6f/js/MLP.js#L213

**Comment by victorqribeiro** [comment]
Hi @alexandrucancescu, MLP was my first attempt at writing a neural network and I was trying to implement Softmax on it (which needs the array of weights and also each individual value. I have probably screw up the code (I don't even remember using Softmax with any examples I made using this NN). As soon as I was finished writing this, I almost immediately moved on to "Dejavu" (https://github.com/victorqribeiro/aimAndShoot/blob/master/js/Dejavu.js) which was my next iteration. I think Dejavu is more robust than MLP but either way both of them are toy neural networks (at least I considered that way).
